pkg/chain/ethereum/bitcoin_difficulty.go

```diff
@@ -0,0 +1,107 @@
+package ethereum
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/keep-network/keep-common/pkg/chain/ethereum"
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
+	"github.com/keep-network/keep-core/pkg/chain/ethereum/tbtc/gen/contract"
+)
+
+// Definitions of contract names.
+const (
+	LightRelayContractName = "LightRelay"
+)
+
+// BitcoinDifficultyChain represents a Bitcoin difficulty-specific chain handle.
+type BitcoinDifficultyChain struct {
+	*baseChain
+
+	lightRelay *contract.LightRelay
+}
+
+// NewBitcoinDifficultyChain construct a new instance of the Bitcoin difficulty
+// - specific Ethereum chain handle.
+func NewBitcoinDifficultyChain(
```

Comment:
Does it have to be exported?

Reply:
It needs to be exported, because it will be used outside this package.
I realised this function can already be called and added the call to it in 6d2f2574c59e0e72ab16c4497a590ef980487f36.